src/components/pwc-choices-2-dropdown/pwc-choices-2-dropdown.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Component,
+  Prop,
+  h,
+  Event,
+  EventEmitter,
+  Element,
+  State
+} from "@stencil/core";
+import { PwcChoices2 } from "../../interfaces/PwcChoices2";
+import _ from "lodash";
+import fuzzy, { FilterResult } from "fuzzy";
+
+@Component({
+  tag: "pwc-choices-2-dropdown",
+  styleUrl: "../styles.scss",
+  shadow: true
+})
+export class PwcChoices2DropdownComponent {
+  @Element() root: HTMLElement;
+
+  @Prop() options: PwcChoices2.IOption[];
+  @Prop() noOptionsString: string;
+
+  @Event() dropdownOptionClicked: EventEmitter<
+    PwcChoices2.IDropdownOptionClickedEventPayload
+  >;
+
+  @State() filteredOptions: Array<FilterResult<PwcChoices2.IOption>>;
+
+  componentWillLoad() {
+    this.filteredOptions = this.makeOptionsWholeFilterResult();
+  }
+
+  render() {
+    return this.constructDropdown();
+  }
+
+  constructDropdown() {
+    return (
+      <div class="dropdown">
+        <input
+          type="text"
+          class="search"
+          placeholder="Search by typing..."
+          onInput={e => this.onSearchInput(e)}
+        ></input>
+        <ul>
+          {this.filteredOptions && this.filteredOptions.length === 0 ? (
+            <li id="noOptionsListItem"> {this.noOptionsString}</li>
+          ) : (
+            this.filteredOptions.map(option =>
+              this.constructDropdownOption(option)
+            )
+          )}
+        </ul>
+      </div>
+    );
+  }
+
+  onSearchInput(e: Event): void {
+    e.stopPropagation();
+    const wholeSearchInput = this.getWholeSearchInput();
+
+    if (wholeSearchInput.length === 0) {
+      this.filteredOptions = this.makeOptionsWholeFilterResult();
```

Comment:
`makeOptionsWholeFilterResult` change it all

Reply:
a17975ce7d50a96f9eaf78d26f7c23366f4f9d5e